grpc/src/test/java/com/linecorp/armeria/internal/client/grpc/CustomGrpcClientFactoryTest.java

```diff
@@ -78,4 +115,72 @@ public Object newClientStub(Class<?> clientType, Channel channel) {
           .hasMessageContaining("Unexpected client stub type: " +
                                 TestServiceGrpc.TestServiceBlockingStub.class.getName());
     }
+
+    @Test
+    void interceptors() {
+        final AtomicInteger invoked1 = new AtomicInteger();
+        final AtomicInteger invoked2 = new AtomicInteger();
+        final ClientInterceptor firstInterceptor = new ClientInterceptor() {
+            @Override
+            public <I, O> ClientCall<I, O> interceptCall(
+                    MethodDescriptor<I, O> method, CallOptions callOptions, Channel next) {
+                return new SimpleForwardingClientCall<I, O>(next.newCall(method, callOptions)) {
+                    @Override
+                    public void start(Listener<O> responseListener, Metadata headers) {
+                        logger.info("first");
+                        invoked1.getAndIncrement();
+                        super.start(responseListener, headers);
```

Comment:
nit; what do you think of removing the logging, and asserting from interceptors?
This way, we can be sure the interceptors are run in the correct order.

ditto for the other interceptors

```suggestion
                        invoked1.getAndIncrement();
                        assertThat(invoked1).hasValue(1);
                        assertThat(invoked2).hasValue(0);
                        assertThat(invoked3).hasValue(0);
                        super.start(responseListener, headers);
```

Reply:
It seems good